### DISCUSSION TITLE
[Auth0-Client] - token not found

### DISCUSSION THREAD

**Discussion by georgev-97** [mia-platform/community#509] [Q&A] [answered]
### What product are you having troubles with?

Console

### What Console version are you using?

12

### Description

Hello,

I am currently using Auth0-client to get access tokens and authorize applications,
the Auth0-client is deployed in preproduction environment and I am reaching the routes /oauth/token and /userinfo from different projects on different environments.

### Actual Outcome

When I get the access token from **microservice X** using the route /oauth/token and then I use that token to get userinfo from a **microservice Y** (a frontend or from postman for example) I get the error `failed token retrieval: access token in redis is empty`

### Expected Outcome

The `GET /userinfo` should return user data if the token is correct.
I've tested the token by calling the auth0 tenant directly and it works as expected.

**Comment by fredmaggiowski** [comment, marked as answer]
Hi @georgev-97, sorry for the late reply!

> When I get the access token from microservice X using the route /oauth/token and then I use that token to get userinfo from a microservice Y (a frontend or from postman for example) I get the error failed token retrieval: access token in redis is empty

Tokens stored by the `auth0-client` are bound to the specific host they have been signed for, coming from different hosts may lead to such error. Does this make sense to troubleshoot your issue?

Did you manage to resolve this anyway?
